src/EnergyPlus/UnitVentilator.cc

```diff
@@ -3048,22 +3048,11 @@ namespace UnitVentilator {
                             OAMassFlowRate = MinOAFrac * Node(OutsideAirNode).MassFlowRate;
 
                         } else if (SELECT_CASE_var == VariablePercent) {
-                            // This algorithm is probably a bit simplistic in that it just bounces
-                            // back and forth between the maximum outside air and the minimum.  In
-                            // REAL(r64)ity, a system *might* vary between the two based on the load in
-                            // the zone.  This simple flow control might cause some overcooling but
-                            // chances are that if there is a cooling load and the zone temperature
-                            // gets above the outside temperature that overcooling won't be significant.
-                            if (Tinlet <= Toutdoor) {
-
-                                OAMassFlowRate = MinOAFrac * Node(OutsideAirNode).MassFlowRate;
-
-                            } else { // Tinlet > Toutdoor
-
-                                MaxOAFrac = GetCurrentScheduleValue(UnitVent(UnitVentNum).MaxOASchedPtr);
-                                OAMassFlowRate = MaxOAFrac * Node(OutsideAirNode).MassFlowRate;
-                            }
-
+                            
+                            OAMassFlowRate = SetOAMassFlowRateForCoolingVariablePercent(MinOAFrac,Node(OutsideAirNode).MassFlowRate,
+                                                                                        GetCurrentScheduleValue(UnitVent(UnitVentNum).MaxOASchedPtr),
+                                                                                        Tinlet,Toutdoor);
```

Comment:
Nice to pull this out into a separate function.

Reply:
Thanks--makes unit testing cleaner as well.